pkg/util/docker/dockerfile/dockerfile.go

```diff
@@ -0,0 +1,55 @@
+package dockerfile
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/docker/docker/builder/parser"
+)
+
+// ParseTreeToDockerfile takes a Dockerfile AST node, as generated by
+// parser.Parse, and returns a new, equivalent, Dockerfile.
+func ParseTreeToDockerfile(node *parser.Node) []byte {
+	buf := []byte(node.Original)
+	for _, child := range node.Children {
+		buf = append(buf, ParseTreeToDockerfile(child)...)
+	}
+	// Append a line break when needed.
+	if len(buf) > 0 && buf[len(buf)-1] != '\n' {
+		buf = append(buf, '\n')
+	}
+	return buf
+}
+
+// FindAll returns the indices of all children of node such that
+// node.Children[i].Value == command.
+func FindAll(node *parser.Node, command string) []int {
```

Comment:
Scratch specifying a position, I meant returning the position of the last of occurence. Commands like CMD, ENTRYPOINT are valid only once in a Dockerfile and that's the last occurence of them.